src/components/main/Main.tsx

```diff
@@ -0,0 +1,147 @@
+import styles from "./Main.module.css";
+
+import axios from "axios";
+
+import {
+  MainProps,
+  CryptoItem,
+  CryptoItemWithPrice,
+} from "../../data/interfaces";
+
+import { useState, useRef, useEffect } from "react";
+
+import InputCryptoCode from "../inputCryptoCode/InputCryptoCode";
+import CryptoPinnedCard from "../cryptoPinnedCard/CryptoPinnedCard";
+import CryptoSuggestCard from "../cryptoSuggestCard/CryptoSuggestCard";
+
+function Main({ wrapperPrimaryRef }: MainProps) {
+  // STATES
+  const [cryptoCode, setCryptoCode] = useState<string>("");
+  const [cryptoList, setCryptoList] = useState<CryptoItem[]>([]);
+  const [cryptoListToSuggest, setCryptoListToSuggest] = useState<CryptoItem[]>(
+    []
+  );
+  const [cryptoListPinned, setCryptoListPinned] = useState<
+    CryptoItemWithPrice[]
+  >([]);
+  const [modalVisibility, setModalVisibility] = useState<boolean>(false);
+
+  // REFS
+  const priceModalRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    // FUNCTION TO UPDDATE PINNED PRICES
+    const updatePinnedPrices = async (
+      pinnedCryptoList: CryptoItemWithPrice[]
+    ) => {
+      const result: Promise<CryptoItemWithPrice>[] = [...pinnedCryptoList].map(
+        async (item: CryptoItemWithPrice, index: number) => {
+          const response: any = await axios.get(
+            `${import.meta.env.VITE_API_CRYPTO_PRICE}${item.id}-usd/spot`
+          );
+          return {
+            ...pinnedCryptoList[index],
+            price: response.data.data.amount,
+          };
+        }
+      );
+      return await Promise.all(result);
+    };
+
+    // CHECK IF THERE ARE PINNED CRYPTOCURRENCIES IN LOCALSTORAGE
+    const retrievedPinnedCryptoList: string | null =
+      window.localStorage.getItem("cryptoListPinned");
+    if (retrievedPinnedCryptoList !== null) {
+      const parsedCryptoList: CryptoItemWithPrice[] = JSON.parse(
+        retrievedPinnedCryptoList
+      );
+      updatePinnedPrices(parsedCryptoList).then(
+        (result: CryptoItemWithPrice[]) => {
+          setCryptoListPinned(result);
+        }
+      );
+    }
+    // API CALL TO FETCH ALL THE AVAILABLE CRYPTOCURRENCIES (NAMES AND CODES)
+    axios
+      .get(import.meta.env.VITE_API_CRYPTO_LIST)
```

Comment:
Try to simplify useEffect by separating API call logic into a fetchData function